test/appender/test_appender.cpp

```diff
@@ -169,6 +169,141 @@ TEST_CASE("Test AppendRow", "[appender]") {
+		REQUIRE_NOTHROW(appender.EndRow());
+		// NOT NULL constraint failed
+		REQUIRE_THROWS(appender.Close());
+	}
+	result = con.Query("SELECT * FROM integers");
+	auto chunk = result->Fetch();
+	REQUIRE(chunk == nullptr);
+
+	// DEFAULT nextval('seq')
+	REQUIRE_NO_FAIL(con.Query("CREATE SEQUENCE seq"));
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE integers(i iNTEGER, j INTEGER DEFAULT nextval('seq'))"));
+	{
+		Appender appender(con, "integers");
+		appender.BeginRow();
+		appender.Append<int32_t>(1);
+		REQUIRE_NOTHROW(appender.AppendDefault());
+		REQUIRE_NOTHROW(appender.EndRow());
+		REQUIRE_NOTHROW(appender.Close());
+	}
+	result = con.Query("SELECT * FROM integers");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::INTEGER(1)}));
+	REQUIRE(CHECK_COLUMN(result, 1, {Value::INTEGER(1)}));
+
+	// DEFAULT random()
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE integers(i iNTEGER, j DOUBLE DEFAULT random())"));
+	con.Query("select setseed(0.42)");
+	{
+		Appender appender(con, "integers");
+		appender.BeginRow();
+		appender.Append<int32_t>(1);
+		REQUIRE_NOTHROW(appender.AppendDefault());
+		REQUIRE_NOTHROW(appender.EndRow());
+		REQUIRE_NOTHROW(appender.Close());
+	}
+	result = con.Query("SELECT * FROM integers");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::INTEGER(1)}));
+	REQUIRE(CHECK_COLUMN(result, 1, {Value::DOUBLE(0.4729174713138491)}));
+
+	// DEFAULT now()
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE integers(i iNTEGER, j TIMESTAMPTZ DEFAULT now())"));
+	con.Query("BEGIN TRANSACTION");
+	result = con.Query("select now()");
+	auto &materialized_result = result->Cast<MaterializedQueryResult>();
+	auto current_time = materialized_result.GetValue(0, 0);
+	{
+		Appender appender(con, "integers");
+		appender.BeginRow();
+		appender.Append<int32_t>(1);
+		REQUIRE_NOTHROW(appender.AppendDefault());
+		REQUIRE_NOTHROW(appender.EndRow());
+		REQUIRE_NOTHROW(appender.Close());
+	}
+	result = con.Query("SELECT * FROM integers");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::INTEGER(1)}));
+	REQUIRE(CHECK_COLUMN(result, 1, {current_time}));
+	con.Query("COMMIT");
+}
+
+TEST_CASE("Test append default into Vector", "[appender]") {
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE integers(i iNTEGER)"));
+
+	auto result = con.Query("SELECT a::INTEGER FROM RANGE(15) t(a)");
+	auto chunk = result->Fetch();
+	D_ASSERT(result->type == QueryResultType::MATERIALIZED_RESULT);
+
+	{
+		Appender appender(con, "integers");
+
+		auto &column = chunk->data[0];
+		SelectionVector sel(3);
+
+		sel.set_index(0, 5);
+		sel.set_index(1, 8);
+		sel.set_index(2, 3);
+
+		appender.AppendDefaultToVector(column, 0, sel, 3);
+		REQUIRE(column.GetValue(0) == 0);
+		REQUIRE(column.GetValue(1) == 1);
+
+		REQUIRE(column.GetValue(5).IsNull());
+
+		REQUIRE(column.GetValue(8).IsNull());
+
+		REQUIRE(column.GetValue(3).IsNull());
+
+		REQUIRE(!column.GetValue(14).IsNull());
+		REQUIRE(column.GetValue(14) == 14);
```

Comment:
Let's extend this test by trying to append the data chunk to the table after setting the default values. Can we also add other `AppendDefaultsToVector` tests triggering the exceptions?